netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java

```diff
@@ -181,8 +183,13 @@ public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
     }
 
     @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+      fail(ctx, cause);
+    }
+
+    @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
-      failBufferedAndClose(ctx);
+      fail(ctx, new Exception("Connection broken while performing protocol negotiation"));
```

Comment:
So the question is which status to use in each case:

1. channelInactive (unavailable?)
2. close (cancelled?)
3. http2 plaintext upgrade failed (failed_precondition?)